src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -110,8 +101,8 @@
     private String charset;
 
     private LogActionImpl(FlowNode parent, Charset charset) {
-        if (!isRunning(parent)) {
-            throw new IllegalStateException("cannot start writing logs to a finished node " + parent);
+        if (!parent.isActive()) {
+            LOGGER.log(Level.WARNING, null, new IllegalStateException("cannot start writing logs to a finished node " + parent + " " + parent.getDisplayFunctionName() + " in " + parent.getExecution()));
```

Comment:
> is there a legit case for instantiating when the parent isn't active?

No.

> warn but no exception thrown

That is what we are doing here.

> throw exception but use try/catch in the flow elsewhere so we don't break the build

Perhaps `DefaultStepContext.get` could throw this as an `IOException`, which in principle `StepExecution`s should already be catching.